Source/Core/Resource.js

```diff
@@ -212,21 +212,35 @@ function combineQueryParameters(q1, q2, preserveQueryParameters) {
   return result;
 }
 
+/** @typedef {string | number | boolean} QueryValue */
+/**
+ * @typedef {Object.<string, QueryValue | QueryValue[] | undefined>} Resource.QueryParameters
+ * QueryParameters must be an object whose keys are parameter names and whose
+ * values are a primitive (string, number, boolean) or an array thereof.
+ * */
+
+/**
+ * @typedef {Object} Resource.ConstructorOptions
+ *
+ * Initialization options for the Resource constructor
+ *
+ * @property {String} url The url of the resource.
+ * @property {Resource.QueryParameters} [queryParameters] An object containing query parameters that will be sent when retrieving the resource.
```

Comment:
I'm not sure why this is needed, especially since there are a lot of other instances of `queryParmeters` in the code that just use `Object`.  It leads to confusing JSDoc

![image](https://user-images.githubusercontent.com/1495196/83669739-0d9d8980-a5a0-11ea-9cf7-6e109b6cb87b.png)

Since we ultimately just `toString` the values, I'm not sure why a special type is needed.

Reply:
I think maybe the previous branch had updated some of the other references, but I lost track of it when these changes were backed out?  Anyway, it's been a minute, but I think this https://github.com/CesiumGS/cesium/issues/6205#issuecomment-364888861 might be the reason why, at least on my own typings, I wanted a more constrained type.  I thought that issue (`{x: [1,2,3]}` turning into `x=1&x=2&x=3` instead of `x=1,2,3`) had actually been resolved, which would mean that the values are not simply `toString`'d.  Could be mistaken?  As a later comment in that thread points out, `Resource` isn't really "for" consumers anyway.

(What really should be getting passed here is [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) but I get that you still have legacy support to consider and it's almost certainly not worth the weight of a polyfill.)

Reply:
Just to be sure, I did some testing and:

```js
var resource = new Resource({
  url: "http://test.invalid/test",
  queryParameters: {
    key1: ["value1", "value5", "value6"],
    key2: "value2",
  }
});
```

Gets turned into `http://test.invalid/test?key1=value1&key1=value5&key1=value6&key2=value2`

And

```js
var someObject = { toString: () => "bob" };
var resource = new Resource({
  url: "http://test.invalid/test",
  queryParameters: {
    key1: someObject,
  },
});
```

Gets turned into: `http://test.invalid/test?key1=bob`

So toString is definitely called, though it's smart enough to loop over arrays and add the key multiple times.  While I appreciate the desire to more clearly define the types, I think having a marker object in the JSDoc HTML is confusing.  We would also need to update the other places where queryPamaters is used, such as `setQueryParamaters`.  I'd rather just back the QueryParameters type altogether and use object.  We can certainly keep the Resource options object you introduced.

If you feel strongly about this, please open a separate issue specifically to discuss QueryParameters, I don't want to see this specific piece of code hold up the rest of the PR.

Thanks!

Reply:
I think I originally created that type in response to the issue I linked above, more as a hint to myself than anything.  Please go ahead and back it out and if I feel the need to place some constraint for myself, I will.